exchanges/gateio/gateio.go

```diff
@@ -1,417 +1,929 @@
+	errInvalidOrderStatus                  = errors.New("invalid order status")
+	errInvalidLoanSide                     = errors.New("invalid loan side, only 'lend' and 'borrow'")
+	errInvalidLoanID                       = errors.New("missing loan ID")
+	errInvalidRepayMode                    = errors.New("invalid repay mode specified, must be 'all' or 'partial'")
+	errInvalidAutoRepaymentStatus          = errors.New("invalid auto repayment status response")
+	errMissingPreviewID                    = errors.New("missing required parameter: preview_id")
+	errChangeHasToBePositive               = errors.New("change has to be positive")
+	errInvalidLeverageValue                = errors.New("invalid leverage value")
+	errInvalidRiskLimit                    = errors.New("new position risk limit")
+	errInvalidCountTotalValue              = errors.New("invalid \"count_total\" value, supported \"count_total\" values are 0 and 1")
+	errInvalidTimeInForce                  = errors.New("invalid time in force value")
+	errInvalidAutoSizeValue                = errors.New("invalid \"auto_size\" value, only \"close_long\" and \"close_short\" are supported")
+	errTooManyOrderRequest                 = errors.New("too many order creation request")
+	errInvalidTimeout                      = errors.New("invalid timeout, should be in seconds At least 5 seconds, 0 means cancel the countdown")
+	errNoTickerData                        = errors.New("no ticker data available")
+	errUnsupportedSettleValue              = errors.New("only 'usd','usdt', and 'btc' quote of currency pairs are supported")
+	errOnlyLimitOrderType                  = errors.New("only order type 'limit' is allowed")
+	errNilArgument                         = errors.New("null argument")
+	errInvalidTimezone                     = errors.New("invalid timezone")
+	errMultipleOrders                      = errors.New("multiple orders passed")
 )
 
 // Gateio is the overarching type across this package
 type Gateio struct {
 	exchange.Base
 }
 
-// GetSymbols returns all supported symbols
-func (g *Gateio) GetSymbols(ctx context.Context) ([]string, error) {
-	var result []string
-	urlPath := fmt.Sprintf("/%s/%s", gateioAPIVersion, gateioSymbol)
-	err := g.SendHTTPRequest(ctx, exchange.RestSpotSupplementary, urlPath, &result)
-	return result, err
-}
+// ***************************************** SubAccounts ********************************
 
-// GetMarketInfo returns information about all trading pairs, including
-// transaction fee, minimum order quantity, price accuracy and so on
-func (g *Gateio) GetMarketInfo(ctx context.Context) (MarketInfoResponse, error) {
-	type response struct {
-		Result string        `json:"result"`
-		Pairs  []interface{} `json:"pairs"`
+// CreateNewSubAccount creates a new sub-account
+func (g *Gateio) CreateNewSubAccount(ctx context.Context, arg SubAccountParams) (*SubAccount, error) {
```

Comment:
You are missing response fields here: https://www.gate.io/docs/developers/apiv4/en/#create-a-new-sub-account